src/SagaRepository.js

```diff
@@ -29,10 +29,7 @@ export default class SagaRepository {
     dispatch
   ) {
     if (!this.sagas[sagaId]) {
-      const sagaInstance = new SagaAbstraction(saga, model);
-      sagaInstance.subscribe(dispatch);
-
-      this.sagas[sagaId] = sagaInstance;
+      this.sagas[sagaId] = new SagaAbstraction(saga, model, dispatch);
```

Comment:
This is the major change, instead of subscribing after the instance is constructed, we'll do it inside constructor.